Bug/issue 322 trade widget

adarshram opened the pull request:
fixes #322 

![image](https://user-images.githubusercontent.com/9465239/142624079-291cf0e2-f349-4da1-86c8-2f1c2d3da96c.png)
![image](https://user-images.githubusercontent.com/9465239/142623914-51e071e8-354f-48c7-a6fb-4018689413d7.png)
market and limit input values carry over . 
There is a known issue i'm not able to fix . if you enter from the limit widget with a from value to market widget the you recieve column is not getting triggered . this is due to the fromDebounce value being false and  its triggered from the TokenInputField element . 
Example screen shot 
![image](https://user-images.githubusercontent.com/9465239/142624219-9945ca77-557a-4fea-a84c-494f3765464a.png)

zavelevsky commented on src/elements/swapLimit/SwapLimit.tsx:
```diff
@@ -112,7 +118,7 @@ export const SwapLimit = ({
       }
     },
-    [fromToken]
+    [fromToken,setFromAmount,setFromAmountUsd]
```
please run prettier

adarshram replied:
done .

adarshram replied:
![image](https://user-images.githubusercontent.com/9465239/143003505-ee104f9c-5026-4cc9-a2e8-b280fed21869.png)
this issue seems to be happening after i merged with main .this only happens when you select a receive token and then swap from market to limit . if its in 'select a token' mode its fine

zavelevsky replied:
Thanks, it was opened today - #437

adarshram replied:
shall i fix that in this ticket . i figured the issue . or should i create a seperate branch for it ?

adarshram commented:
Fixed the debounce issue mentioned in the ticket as well . Just added the fromAmount to the debounce and seems like it fixes the issue (instead of doing useDebounce(''))

zavelevsky commented:
Note that in the issue (#322) there's also this requirement:
CTA text to connect a wallet
Did you address it?

adarshram commented:
> Note that in the issue (#322) there's also this requirement: CTA text to connect a wallet Did you address it?

no i havent addressed this .

zavelevsky commented on src/elements/modalDuration/modalDuration.tsx:
```diff
@@ -28,23 +28,23 @@ export const ModalDuration = ({
   const daysItems: DurationItem[] = Array.from(
     { length: 31 },
-    (index: number) => ({
+    (_,index: number) => ({
```
this fix was done already and it causes you conflicts. Also, again, please set prettier to automatically format please.

adarshram replied:
i resolved the conflict now ..

zavelevsky commented:
@adarshram  I will merge your PR to another branch and create a PR from there. I have to do it because of a cloudflare limitation that doesn't allow me to generate preview links from an external PR. I want to share a preview link with our Product manager so that he can review your fix. So thanks for the contribution - this is just a technical step and then your code will be merged to main in this somewhat over-complex manner ¯\_(ツ)_/¯

adarshram commented:
Awesome thanks I'm learning a lot from this . Will be fixing more over
time after learning some more react programming

On Tue, Nov 23, 2021, 17:57 Doron Zavelevsky ***@***.***>
wrote:

> @adarshram <https://github.com/adarshram> I will merge your PR to another
> branch and create a PR from there. I have to do it because of a cloudflare
> limitation that doesn't allow me to generate preview links from an external
> PR. I want to share a preview link with our Product manager so that he can
> review your fix. So thanks for the contribution - this is just a technical
> step and then your code will be merged to main in this somewhat
> over-complex manner ¯_(ツ)_/¯
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bancorprotocol/webapp-v2/pull/422#issuecomment-976463371>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACIG3F75DYGCOMJR5XOO36LUNOCE3ANCNFSM5IMBSAVA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

zavelevsky commented:
> 

Nice! I'd like that and I'm glad to hear.

zavelevsky reviewed (approved):
LGTM

zavelevsky commented:
@adarshram issue (I'll fix over the new PR) - button text is correct but the button is disabled

adarshram commented:
Okay alright . Will take a look tomorrow

On Tue, Nov 23, 2021, 18:25 Doron Zavelevsky ***@***.***>
wrote:

> @adarshram <https://github.com/adarshram> issue (I'll fix over the new
> PR) - button text is correct but the button is disabled
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bancorprotocol/webapp-v2/pull/422#issuecomment-976490926>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACIG3F5HT6YGCBEBPUG4IE3UNOFNLANCNFSM5IMBSAVA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
